src/vmm/src/builder.rs

```diff
@@ -840,6 +861,11 @@ pub mod tests {
         block_files
     }
 
+    pub(crate) fn insert_boot_timer_device(vmm: &mut Vmm, request_ts: TimestampUs) {
+        let res = attach_boot_timer_device(vmm, request_ts);
+        assert!(res.is_ok());
+    }
```

Comment:
instead of using this function, why not just do `attach_boot_timer_device(vmm, request_ts).unwrap()`?

Reply:
I think it was because it follows the pattern of the other devices,  and makes it visible to the device_manger `persist.rs` tests.  I did try to remove it, but it then requires the `attach_boot_timer_device` function to be made public.

Reply:
> but it then requires the attach_boot_timer_device function to be made public.

It should work with `pub(crate)`.